lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -53,49 +244,60 @@ define((require, exports, module) ->
       { type: "rparen", value: "}" }
     ]
 
+    # empty constructor
     constructor: (@editor) ->
 
-    equalTokens: (token1, token2) ->
-      if token1? and token2?
-        return token1.type == token2.type and token1.value == token2.value
-      else return if token1? or token2? then false else true
+    @equalTokens: (token1, token2) ->
+      return token1.type == token2.type and token1.value == token2.value
 
     getEquationStart: (tokenIterator) ->
+      # if tokenIterator initialy is on the empty line, it's current token is null
+      if not tokenIterator.getCurrentToken()?
```

Comment:
but it might as well mean that iterator is already positioned at the end of iteration, e.g. after getEquationEnd call. I believe that nothing goes wrong because when this iterator is at the end of iteration, it can be safely stepForward'ed arbitrary number of times, but still this code looks suspicious. Can you write a few tests which will call this method with iterator positioned at null (both when it is empty line and end of iteration) and tests equations with empty lines inside?